### ISSUE TITLE
IndexError: List Index Out of Range

### ISSUE THREAD

**Issue by ixl911ixl** [BoBeR182/Pokemon-Go-Bot-Working-Hack-API#17]
2016-07-27 17:41:05,282 [   pokebot] [ERROR] Main loop has an ERROR, restarting list index out of range
Traceback (most recent call last):

  File "C:\Users\(myuserhere)\Desktop\PokeGoBot\pokebot.py", line 95, in main
    api.main_loop()

  File "C:\Users\(myuserhere)\Desktop\PokeGoBot\pgoapi\pgoapi.py", line 510, in main_loop
    if self.pokeballs[self._pokeball_type] > 1:#make sure we always have at least 10 pokeballs, 5 great balls, and 5 ultra balls

IndexError: list index out of range

The game keeps immediately crashing upon checking the inventory count. I don't have 5 ultra balls, is this causing it to bug out? How can I remove this and let it function without that? 

Additionally, it thinks I have "less than 1 pokeball, headed to pokestop!" over and over and over.

**Comment by BoBeR182** [comment]
Do you have any pokeballs, normal great or ultra, the comment on that code is outdated, make sure you are running the latest release.

**Comment by oceaniity** [comment]
I am experiencing this too.
```bash
2016-07-28 14:58:21,776 [    pgoapi] [ INFO] List of Pokemon:
Pidgeotto, CP 269, IV 35.56
Total Pokemon count: 1
2016-07-28 14:58:21,778 [    pgoapi] [ INFO] List of items:
Item_ID 1       item count 13   (POKE_BALL)
Item_ID 101     item count 1    (POTION)
Item_ID 201     item count 11   (REVIVE)
Item_ID 701     item count 10   (RAZZ_BERRY)
Item_ID 901     item count 1    (INCUBATOR_BASIC_UNLIMITED)
Item_ID 902     item count 1    (INCUBATOR_BASIC)
Total item count: 37
2016-07-28 14:58:22,812 [    pgoapi] [ INFO] Less than 1 Poke Balls: Entering pokestops only
```
The above eventually reaches the point where it tries to validate Poke balls, great balls, and ultra balls, and fails:
```python
Traceback (most recent call last):
  File "pokebot.py", line 95, in main
    api.main_loop()
  File "C:\Users\_\Repos\Pokemon-Go-Bot-Working-Hack-API\pgoapi\pgoapi.py", line 489, in main_loop
    if self.pokeballs[self._pokeball_type] > 1: # make sure we always have at least 10 pokeballs, 5 great balls, and 5 ultra balls
IndexError: list index out of range
```

Seems like it's looking up a ball type that does not exist. Makes sense given I have no Great/Ultra balls. Reading the source it looks like the 'Less than 1 pokeballs' message is not accurate:
```python
if self.pokeballs[self._pokeball_type] > 1: # make sure we always have at least 10 pokeballs, 5 great balls, and 5 ultra balls
    while self.catch_near_pokemon():
        sleep(4) # If you want to make it faster, delete this line... would not recommend though
        pass
else:
    self._pokeball_type += 1
self.log.info("Less than 1 Poke Balls: Entering pokestops only")
```

I think a try/except would be appropriate here. You want it to fail if it does not have adequate pokeballs but not completely exit if it has enough *pokeballs*, even in the event others are non-existent. Something like:
```python
try:
    if self.pokeballs[self._pokeball_type] > 1:
        while self.catch_near_pokemon():
            sleep(4)
            pass
    else:
        self.log.info("Less than 1"+self._pokeball_type+" Balls")
        self._pokeball_type += 1
except IndexError: #dont have the pokeball type that was referenced
    self.log.info('Did not have any balls of type ' + self._pokeball_type)
finally:
    self.log.info('Reverted to looting stops.')
    self.spin_near_fort()
```

**Comment by BoBeR182** [comment]
Nice thanks for the work, i think I fixed it in 94ae1f081b828c0e6a6bf90bf2b5231c56638d51

My code always feels very sloppy so feel free to refactor.
